src/lib/telemetry.rs

```diff
@@ -29,10 +30,102 @@ pub enum TraceType {
+    BatchRefundsUpdate,
+    BatchRefundsUpdateFailed,
+    CheckRefundsDeserializeBigQuery,
+    CheckRefundsDeserializeBigQueryFailed,
+    CheckRefundsSubscriptionMissingFromDatabase,
+    CheckRefundsRefundDataChanged,
+    CheckRefundsRefundDataUnchanged,
+    CheckRefundsRefundUpdate,
+    CheckRefundsRefundUpdateFailed,
+    CheckRefundsRefundCreate,
+    CheckRefundsRefundCreateDuplicateKeyViolation,
+    CheckRefundsRefundCreateDatabaseError,
+    CheckRefundsRefundCreateFailed,
+    CheckRefundsRefundFetchFailed,
+    CheckSubscriptionsDeserializeBigQuery,
+    BatchRefundsStarting,
+    BatchRefundsEnding,
+    BatchRefundsTimer,
+    CheckRefundsStarting,
+    CheckRefundsEnding,
+    CheckRefundsTimer,
+    CheckSubscriptionsStarting,
+    CheckSubscriptionsEnding,
+    CheckSubscriptionsTimer,
+    CleanupStarting,
+    CleanupEnding,
+    CleanupTimer,
+    ReportSubscriptionsStarting,
+    ReportSubscriptionsEnding,
+    ReportSubscriptionsTimer,
+    WebAppStarting,
+    WebAppEnding,
+    WebAppTimer,
+    TestStarting,
+    TestEnding,
+    TestTimer,
+    CheckRefundsNFromBq,
+    CheckSubscriptionsNFromBq,
+    CheckRefundsTotalNFromBq,
+    CheckSubscriptionsTotalNFromBq,
+    CheckRefundsBytesFromBq,
+    CheckSubscriptionsBytesFromBq,
+    ReportSubscriptionsNNotReported,
+    TestIncr,
+    TestGauge,
+    TestTime,
+    BatchRefundsNNotReported,
+    CorrectionsReportByDayAccessed,
+    CorrectionsReportTodayAccessed,
+    CheckSubscriptionsDeserializeBigQueryFailed,
+    CheckSubscriptionsAicFetch,
+    CheckSubscriptionsAicFetchFromArchive,
+    CheckSubscriptionsAicFetchFailed,
+    CheckSubscriptionsAicArchive,
+    CheckSubscriptionsAicArchiveFailed,
+    CheckSubscriptionsSubscriptionCreate,
+    CheckSubscriptionsSubscriptionCreateDuplicateKeyViolation,
+    CheckSubscriptionsSubscriptionCreateDatabaseError,
+    CheckSubscriptionsSubscriptionCreateFailed,
+    CleanupAicArchive,
+    CleanupAicArchiveFailed,
+    ReportSubscriptionsAicExpiredBeforeSubscriptionCreated,
+    ReportSubscriptionsSubscriptionHasNoAicExpiry,
+    ReportSubscriptionMarkWillNotReport,
+    ReportSubscriptionMarkWillNotReportFailed,
+    ReportSubscriptionReportToCj,
+    ReportSubscriptionReportToCjButCouldNotMarkReported,
+    ReportSubscriptionReportToCjFailed,
+    ReportSubscriptionMarkNotReported,
+    ReportSubscriptionMarkNotReportedFailed,
+    StatusHistoryDeserializeError,
+}
+
+// TODO doc
+// TODO test
+// TODO doc weird nit about suffixes and what happens when they fail
+impl LogKey {
+    pub fn add_suffix(&self, suffix: &str) -> LogKey {
+        let s = format!("{}-{}", &self.to_string(), suffix);
+        let s_str = s.as_str();
+
+        match LogKey::from_str(s_str) {
+            Ok(v) => v,
+            // TODO why can't I just return self ...
+            // TODO do we need to deal with the unwrap? Hope not
+            Err(_) => LogKey::from_str(&self.to_string()).unwrap(),
```

Comment:
Not sure why, but I can't figure out a way to just return `self` or `&self`. This was my best attempt. But now it's ugly and also we have this `unwrap` here.

Reply:
I've added tests for it so we can be convinced that it's working properly.

I know what you mean about the unwrap being weird but I do understand why we need it and I think it's ok. I tried tweaking once I wrote the test and couldn't figure it out. We can ask one of the rust experts for a consult for curiosity and growth.